File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        mp = defaultdict(list)

        for num in arr:
            mp[bin(num).count("1")].append(num)

        ans = []

        for k in sorted(mp.keys()):
            for i in sorted(mp[k]):
                ans.append(i)

        return ans

File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.cs
public class Solution {
    public int[] SortByBits(int[] arr) {
        return arr.OrderBy(x => CountOnes(x)).ThenBy(x => x).ToArray();
    }

    private int CountOnes(int x) {
        int count = 0;
        
        while (x != 0) {
            count += x & 1;
            x >>= 1;
        }
        return count;
    }
}